From 6d284f1ff81494a5fca91a399b92b218ea1a9ea8 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Tue, 8 Jun 2021 08:53:30 -0700
Subject: [PATCH] Handle enum element override

NETPLAN_DEF_TYPE_VIRTUAL and NETPLAN_DEF_TYPE_BRIDGE point
to same value in enum, however here they are assigned individually
which results in overriding the initialization of the objects

Fixes
src/netplan.h:85:33: error: initializer overrides prior initialization of this subobject [-Werror,-Winitializer-overrides]
    [NETPLAN_DEF_TYPE_BRIDGE] = "bridges",
                                ^~~~~~~~~
Upstream-Status: Submitted [https://github.com/canonical/netplan/pull/213]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 src/netplan.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/netplan.h b/src/netplan.h
index d2c538b..26574f8 100644
--- a/src/netplan.h
+++ b/src/netplan.h
@@ -81,7 +81,6 @@ static const char* const netplan_def_type_to_str[NETPLAN_DEF_TYPE_MAX_] = {
     [NETPLAN_DEF_TYPE_ETHERNET] = "ethernets",
     [NETPLAN_DEF_TYPE_WIFI] = "wifis",
     [NETPLAN_DEF_TYPE_MODEM] = "modems",
-    [NETPLAN_DEF_TYPE_VIRTUAL] = NULL,
     [NETPLAN_DEF_TYPE_BRIDGE] = "bridges",
     [NETPLAN_DEF_TYPE_BOND] = "bonds",
     [NETPLAN_DEF_TYPE_VLAN] = "vlans",
-- 
2.32.0

